src/ensureRefIsUsedByChild.js

```diff
@@ -0,0 +1,19 @@
+import isDOMElement from './isDOMElement';
+
+export const errorMessage =
+  '<Waypoint> needs a DOM element to compute boundaries. The child you passed is neither a ' +
+  'DOM element (e.g. <div>) nor does it uses innerRef prop.\n\n' +
+  'See /* TODO: shortlink */ for more info.';
+
+/**
+ * Raise an error if "children" is not a DOM Element and there is no ref provided to Waypoint
+ *
+ * @param {?(React.element|Function)} children
+ * @param {?HTMLElement} ref
+ * @return {undefined}
+ */
+export default function ensureRefIsProvidedByChild(children, ref) {
+  if (children && !isDOMElement(children) && !ref) {
```

Comment:
Do we need to check for DOM element here? Would checking the existence of `ref` be enough?

Reply:
If a child is a DOM Element and there is no `ref` in `Waypoint` (although this probably could never happen, I'm not sure) this is probably some other issue and not the passing of Component as a child

The current error message could confuse the user, it states:

> The child you passed is neither a DOM element (e.g. \<div\>) ...

That is why I added this check. We could remove it but in some weird case, error message could mislead someone :)

Reply:
👍